Source/Extensions/ConstraintsSetup.swift

```diff
@@ -123,9 +123,32 @@ extension ImagePickerController {
         multiplier: 1, constant: 0))
     }
 
-    view.addConstraint(NSLayoutConstraint(item: bottomContainer, attribute: .height,
-      relatedBy: .equal, toItem: nil, attribute: .notAnAttribute,
-      multiplier: 1, constant: BottomContainerView.Dimensions.height))
+    if #available(iOS 11.0, *) {
+      view.addConstraint(NSLayoutConstraint(item: topView, attribute: .top,
+                                            relatedBy: .equal, toItem: view.safeAreaLayoutGuide,
+                                            attribute: .top,
+                                            multiplier: 1, constant: 0))
+    } else {
+      view.addConstraint(NSLayoutConstraint(item: topView, attribute: .top,
+                                            relatedBy: .equal, toItem: view,
+                                            attribute: .top,
+                                            multiplier: 1, constant: 0))
+    }
+    
+    if #available(iOS 11.0, *) {
+      let heightPadding = UIApplication.shared.keyWindow!.safeAreaInsets.bottom
+      view.addConstraint(NSLayoutConstraint(item: bottomContainer, attribute: .height,
+                                            relatedBy: .equal, toItem: nil,
+                                            attribute: .notAnAttribute,
+                                            multiplier: 1,
+                                            constant: BottomContainerView.Dimensions.height + heightPadding))
+    } else {
+      view.addConstraint(NSLayoutConstraint(item: bottomContainer, attribute: .height,
+                                            relatedBy: .equal, toItem: nil,
+                                            attribute: .notAnAttribute,
+                                            multiplier: 1,
+                                            constant: BottomContainerView.Dimensions.height))
+    }
```

Comment:
Nice! I was also about to add the iPhoneX support, but I'm glad that you were quicker ;-)

LGTM 👍

Just as an idea: You could add the new constraints in a bit more compact way like this:

```
    var topViewToItem: Any? = view
    var bottomContainerHeight = BottomContainerView.Dimensions.height
    
    if #available(iOS 11.0, *) {
        topViewToItem = view.safeAreaLayoutGuide
        bottomContainerHeight += UIApplication.shared.keyWindow!.safeAreaInsets.bottom
    }
    
    view.addConstraint(NSLayoutConstraint(item: topView, attribute: .top,
                                          relatedBy: .equal, toItem: topViewToItem,
                                          attribute: .top,
                                          multiplier: 1, constant: 0))
    
    view.addConstraint(NSLayoutConstraint(item: bottomContainer, attribute: .height,
                                            relatedBy: .equal, toItem: nil,
                                            attribute: .notAnAttribute,
                                            multiplier: 1,
                                            constant: bottomContainerHeight)
    )
```